src/main/java/dansplugins/factionsystem/eventhandlers/DamageHandler.java

```diff
@@ -106,40 +106,77 @@ private void handleEntityDamage(Player attacker, EntityDamageByEntityEvent event
+//            event.setCancelled(true);
+//        }
     }
 
-    private void cancelDamageIfNecessary(EntityDamageByEntityEvent event, Faction playersFaction) {
-        ClaimedChunk claimedChunk = getClaimedChunkAtLocation(event.getEntity().getLocation());
-        if (claimedChunk == null) {
-            return;
-        }
-
-        if (!isHolderPlayersFaction(claimedChunk, playersFaction)) {
-            event.setCancelled(true);
-        }
-    }
+//    private void cancelDamageIfNecessary(EntityDamageByEntityEvent event, Faction playersFaction) {
+//        ClaimedChunk claimedChunk = getClaimedChunkAtLocation(event.getEntity().getLocation());
+//        if (claimedChunk == null) {
+//            return;
+//        }
+//
+//        if (!isHolderPlayersFaction(claimedChunk, playersFaction)) {
+//            event.setCancelled(true);
+//        }
+//    }
 
-    private boolean isHolderPlayersFaction(ClaimedChunk claimedChunk, Faction playersFaction) {
-        return claimedChunk.getHolder().equalsIgnoreCase(playersFaction.getName());
-    }
+//    private boolean isHolderPlayersFaction(ClaimedChunk claimedChunk, Faction playersFaction) {
+//        return claimedChunk.getHolder().equalsIgnoreCase(playersFaction.getName());
+//    }
 
     private ClaimedChunk getClaimedChunkAtLocation(Location location) {
         Chunk chunk = location.getChunk();
         return persistentData.getChunkDataAccessor().getClaimedChunk(chunk);
     }
 
-    private boolean isEntityProtected(Entity entity) {
-        return entity instanceof ArmorStand || entity instanceof ItemFrame;
+    private boolean isEntityProtected(Player attacker, Entity entity) {
+        if (entity instanceof ArmorStand || entity instanceof ItemFrame) return true;
+
+        // If entity isn't on claimed chunk, return false
+        if (!persistentData.getChunkDataAccessor().isClaimed(entity.getLocation().getChunk())) {
+            logger.debug("Chunk isn't claimed");
+            return false;
+        }
+
+        final Faction chunkHolder = persistentData.getFaction(persistentData.getChunkDataAccessor().getClaimedChunk(entity.getLocation().getChunk()).getHolder());
+
+        if (!(boolean) chunkHolder.getFlags().getFlag("enableMobProtection")) {
+            logger.debug("Mob Protection is disabled");
+            return false;
+        }
 
+        // If entity isn't a living one (like minecarts), return false
+        if (!(entity instanceof LivingEntity)) {
+            logger.debug("Entity isn't a living");
+            return false;
+        }
+
+        // If attacker is in faction which owns the chunk, return false
+        if (persistentData.getPlayersFaction(attacker.getUniqueId()) == chunkHolder) {
+            logger.debug("Attacker is in same faction as Chunkholder");
+            return false;
+        }
+
+        // If attacker is factionless, return true
+        if (persistentData.getFaction(attacker.getName()) == null) {
```

Comment:
It's probably worth pulling this out into a local variable rather than calling getFaction twice.
This avoids making the assumption that getFaction is and will always be idempotent and only callable from a single thread.

Reply:
done